extensions/arc/src/ui/dashboards/miaa/miaaComputeAndStoragePage.ts

```diff
@@ -0,0 +1,371 @@
+			value: loc.computeAndStorageDescriptionPartOne,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px', 'max-width': 'auto' }
+		}).component();
+		const infoComputeStorage_p2 = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.miaaComputeAndStorageDescriptionPartTwo,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const memoryVCoreslink = this.modelView.modelBuilder.hyperlink().withProperties<azdata.HyperlinkComponentProperties>({
+			label: loc.scalingCompute,
+			url: 'https://docs.microsoft.com/azure/azure-arc/data/configure-managed-instance',
+			CSSStyles: { 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const infoComputeStorage_p4 = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.computeAndStorageDescriptionPartFour,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const infoComputeStorage_p5 = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.computeAndStorageDescriptionPartFive,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const infoComputeStorage_p6 = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.computeAndStorageDescriptionPartSix,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const computeInfoAndLinks = this.modelView.modelBuilder.flexContainer().withLayout({ flexWrap: 'wrap' }).component();
+		computeInfoAndLinks.addItem(infoComputeStorage_p1, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(infoComputeStorage_p2, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(memoryVCoreslink, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(infoComputeStorage_p4, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(infoComputeStorage_p5, { CSSStyles: { 'margin-right': '5px' } });
+		computeInfoAndLinks.addItem(infoComputeStorage_p6, { CSSStyles: { 'margin-right': '5px' } });
+		content.addItem(computeInfoAndLinks, { CSSStyles: { 'min-height': '30px' } });
+
+		this.configurationContainer = this.modelView.modelBuilder.divContainer().component();
+		this.configurationContainer.addItems(this.createUserInputSection(), { CSSStyles: { 'min-height': '30px' } });
+		content.addItem(this.configurationContainer, { CSSStyles: { 'margin-top': '30px' } });
+
+		this.initialized = true;
+
+		return root;
+	}
+
+	protected get toolbarContainer(): azdata.ToolbarContainer {
+		// Save Edits
+		this.saveButton = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
```

Comment:
A private createButton method that takes buttonProperties as the first parameter and onDidClick callback function as 2nd parameter would be useful and make the code more concise and readable.

Reply:
A similar approach for 'Text' InputBoxComonent creation with onTextChanged callback registration could be useful.

Reply:
Not really seeing the value in that since it's just a single chained call. If we were doing more logic like hooking up handlers or something that might be more worth it.